lib/Db/LocalMessageMapper.php

```diff
@@ -130,7 +130,8 @@ public function findDue(int $time): array {
 			->from($this->getTableName())
 			->where(
 				$qb->expr()->isNotNull('send_at'),
-				$qb->expr()->lte('send_at', $qb->createNamedParameter($time, IQueryBuilder::PARAM_INT), IQueryBuilder::PARAM_INT)
+				$qb->expr()->lte('send_at', $qb->createNamedParameter($time, IQueryBuilder::PARAM_INT), IQueryBuilder::PARAM_INT),
+				$qb->expr()->isNull('failed')
```

Comment:
Yeah I was also not sure. Because now we could have a `false` for mysql, only oracle will silently convert `false` to a `null`.

Maybe it has to be or(isNull('failed'), eq('failed', false))

Reply:
> or(isNull('failed'), eq('failed', false))

I would say this one.